### PR TITLE
feat(perf): Add OpenTelemetry docs for Go

### PR DESCRIPTION
**Pull request by tonyo**
Docs for the initial OpenTelemetry support in Go (starting from to-be-released version 0.18.0).

Related to https://github.com/getsentry/sentry-go/issues/486

### CODE REVIEW

File: src/platform-includes/performance/opentelemetry-install/go.mdx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+The Sentry Go OpenTelemetry integration requires `go.opentelemetry.io/otel` (and its submodules) of version `1.11.0` or higher.
+
+</Note>
+
+Install the `otel` module in addition to the main SDK:
+
+```bash
+go get github.co/getsentry/sentry-go \
+       github.co/getsentry/sentry-go/otel
```
Comment (cleptric) [inline src/platform-includes/performance/opentelemetry-install/go.mdx:17]:
```suggestion
go get github.co/getsentry/sentry-go@latest \
       github.co/getsentry/sentry-go/otel@latest
```

Maybe?

Comment (tonyo) [reply src/platform-includes/performance/opentelemetry-install/go.mdx:17]: Tbh, not exactly sure what is the difference?

Comment (tonyo) [reply src/platform-includes/performance/opentelemetry-install/go.mdx:17]: But `github.co` is :)

**Comment by tonyo** [comment]
Added a section explaining how to link errors and transactions.
That will be improved after https://github.com/getsentry/sentry-go/issues/555 is fixed.

**Review by lizokm — APPROVED** [review]
Small tweak, but mostly LGTM

File: src/platform-includes/performance/opentelemetry-setup/go.mdx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+Now, all spans produced by OpenTelemetry will be also captured and sent to Sentry.
+
+### Linking Errors to Transactions
+
+If you want the errors and messages the SDK produces (for example, via `CaptureException` or
+`CaptureMessage`) to be linked to transactions in the UI, there are a couple extra steps required
+right now. Most importantly, the current OpenTelemetry-enhanced context has to be passed as part
+of `EventHint` to `client.Capture*` methods:
```
[the remark this answers is not in the document]
Comment (tonyo) [reply src/platform-includes/performance/opentelemetry-setup/go.mdx]: Thanks!

File: src/platform-includes/performance/opentelemetry-install/go.mdx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<Note>
+
+OpenTelemetry integration was added in `sentry-go` version 0.18.0, and works only for Go >= 1.18.
+
+</Note>
+
+<Note>
+
+The Sentry Go OpenTelemetry integration requires `go.opentelemetry.io/otel` (and its submodules) of version `1.11.0` or higher.
+
```
Comment (lizokm) [inline src/platform-includes/performance/opentelemetry-install/go.mdx:10]:
```suggestion
The Sentry Go OpenTelemetry integration requires `go.opentelemetry.io/otel` (and its submodules), version `1.11.0` or higher.

```

File: src/platform-includes/performance/opentelemetry-setup/go.mdx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+})
+
+tp := sdktrace.NewTracerProvider(
+	sdktrace.WithSpanProcessor(sentryotel.NewSentrySpanProcessor()),
+)
+otel.SetTracerProvider(tp)
+otel.SetTextMapPropagator(sentryotel.NewSentryPropagator())
+```
+
+Now, all spans produced by OpenTelemetry will be also captured and sent to Sentry.
```
Comment (lizokm) [inline src/platform-includes/performance/opentelemetry-setup/go.mdx:27]:
```suggestion
Now, all spans produced by OpenTelemetry will also be captured and sent to Sentry.
```

**Review by lizokm — APPROVED** [review]
Made a few minor tweaks.

**Pull request merged**